src/lib/esnext.promise.d.ts

```diff
@@ -0,0 +1,15 @@
+interface PromiseConstructor {
+    /**
+     * Takes a callback of any kind (returns or throws, synchronously or asynchronously) and wraps its result
+     * in a Promise.
+     *
+     * @param callbackFn A function that is called synchronously with no arguments. It can do anything: either return
+     * a value, throw an error, or return a promise.
+     *
+     * @returns A Promise that is:
+     * - Already fulfilled, if the callback synchronously returns a value.
+     * - Already rejected, if the callback synchronously throws an error.
+     * - Asynchronously fulfilled or rejected, if the callback returns a promise.
+     */
+    try<T>(callbackFn: () => T | PromiseLike<T>): Promise<T>;

```

Comment:
this is missing the extra args, maybe something like
```suggestion
    try<T>(callbackFn: () => T | PromiseLike<T>, ...args: Parameters<T>): Promise<T>;
```
?

Reply:
Interesting, I just noticed the `args` in the specs (it's still undocumented on MDN).

I'm not very good at reading specs, what are these args for? And when looking at the built-in `Parameters` type, that extracts arguments from a function type, it raises even more questions.

When trying in the latest version of Chrome, I cannot notice any difference.

Reply:
They're passed into the callback.

so i guess instead, it'd be something like:
```suggestion
    try<T, U = unknown[]>(callbackFn: (...args: U) => T | PromiseLike<T>): Promise<T>;
```
?

Reply:
I think in this case you probably want `try<T, U extends any[]>(callbackFn: (...args: U) => T | PromiseLike<T>, ...args: U): Promise<T>`.

Reply:
why `any`? `unknown` represents every possible value.

Reply:
I went for:
```typescript
try<T, U extends unknown[]>(callbackFn: (...args: U) => T | PromiseLike<T>, ...args: U): Promise<T>;
```

Do note that this will throw an type error if you pass too many arguments, which is technically not wrong. But of course it's a good hint to remove the unneeded parameter(s).

Reply:
Indeed, I see that as a feature :-)